### ISSUE TITLE
Chalice should warn or error if generating a name that is too long for CloudFormation

### ISSUE THREAD

**Issue by KeynesYouDigIt** [aws/chalice#1625]
We wrote a handler on a blue print that looked something like 
```
@our_cool_blueprint.schedule(Rate(12, unit=Rate.HOURS))
def rate_obj_handler(event: Cron) -> None:
    do stuff()
```

Which seemed innocent enough. When we went to do a CloudFormation deploy from the results of `chalice package`, cloudformation tried to create this resource

`OurCoolrateObjHandlerOurCoolrateObjHandlerEventLambdaTarget` which encountered this error
```
1 validation error detected: Value 'OurCoolrateObjHandlerOurCoolrateObjHandlerEventLambdaTarget' at 'targets.1.member.id' failed to satisfy constraint: Member must have length less than or equal to 64 ...
```

Could there be a warning or error on chalice package when a resource name is potentially too long? And/or is there something in our naming/blueprint conventions we are doing wrong?

**Comment by jamesls** [comment]
Yeah we can add a warning about a resource name being too long for cloudformation.  I don't see anything wrong with your naming convention.  I'm guessing the longer name comes from specifying a `name_prefix` in your blueprint registration.  Otherwise it should just be `RateObjHandler` and `RateObjHandlerEvent` for the generated names.
